### PR TITLE
Added support in recurring group name in files.

### PR DESCRIPTION
**Pull request by yochaytz**

### CODE REVIEW

**Comment by AKuederle** [comment]
Thanks for the merge request!

In general i think this is a feasible workaround. the only thing I am thinking about is, if it would make sense to change the naming schema for the duplicated groupnames. Maybe {groupname}_{group_id}? How does the groupid usually look like?

**Comment by AKuederle** [comment]
@MattiasFredriksson does this interfere with #43 in any way?

File: c3d/c3d.py [hunk 1]
```diff
@@ -1115,7 +1115,18 @@ def add_group(self, group_id, name, desc):
...
+            #To allow using such files we auto-generate new name. 
+            #Notice that refering to this group's parameters later with the original name will fail.
+            i = 1
+            new_name = name + str(i)
+            while new_name in self._groups:
+                i += 1
+                new_name = name + str(i)
+            warnings.warn('Repeated group name %s modified to %s' % (name, new_name))
+            name = new_name
+            #raise KeyError(name)
```
Comment (friggog) [inline c3d/c3d.py:1128]: Should remove this

File: c3d/c3d.py [hunk 2]
```diff
@@ -1115,7 +1115,18 @@ def add_group(self, group_id, name, desc):
             raise KeyError(group_id)
         name = name.upper()
         if name in self._groups:
-            raise KeyError(name)
+            #In some cases group name is not unique (though c3d spec requires that).
+            #To allow using such files we auto-generate new name. 
+            #Notice that refering to this group's parameters later with the original name will fail.
```
Comment (friggog) [inline c3d/c3d.py:1120]: Comments should start "# "
```suggestion
            # In some cases group name is not unique (though c3d spec requires that).
            # To allow using such files we auto-generate new name. 
            # Notice that referring to this group's parameters later with the original name will fail.
```

File: c3d/c3d.py [hunk 3]
```diff
@@ -1115,7 +1115,18 @@ def add_group(self, group_id, name, desc):
...
+            i = 1
+            new_name = name + str(i)
+            while new_name in self._groups:
+                i += 1
+                new_name = name + str(i)
+            warnings.warn('Repeated group name %s modified to %s' % (name, new_name))
```
Comment (friggog) [inline c3d/c3d.py:1126]: Prefer f-strings
```suggestion
            warnings.warn(f'Repeated group name {name} modified to {new_name}')
```

**Comment by MattiasFredriksson** [comment]
@AKuederle [AKuederle](https://github.com/AKuederle) There should be a conflict but there should be no functional differences (updated the KeyError message to be descriptive).

Otherwise I'm wondering if the name update should be done in this fashion since this is a public method. Not throwing KeyErrors when external code adds a Group could cause silent errors/bugs as it would allow duplicates to be created 'unintentionally'. However, resolving such issues could be done in the __future__ as it would probably complicate the solution a bit.

Currently the standard string format should be on the form:
`
'Repeated group name {} modified to {}'.format(name, new_name)`

**Comment by AKuederle** [comment]
I agree! My suggestion would be to hide the overwrite behind a parameter that is False by default

**Comment by AKuederle** [comment]
I made the changes I suggested above:

1. Made the overwrite optional
2. Changed the rename schmema to be more predictable

@friggog What do you think?

**Review by friggog — APPROVED** [review]

**Comment by AKuederle** [comment]
@yochaytz Thanks again for the contribuition! I will try to make a new release later today containing this fix.

And @friggog and @MattiasFredriksson thanks for the reviews and the input!

**Pull request merged**